internal/provider/resource_tfe_oauth_client.go

```diff
@@ -111,6 +111,11 @@ func resourceTFEOAuthClient() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+
+			"agent_pool_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+			},
```

Comment:
We're missing a few fields here, but I'm not fully familiar with the internals of the OAuth client API:
If the user must specify this attribute in their configuration, we'll need `Require: true`, otherwise `Optional: true`. If this agent pool ID can be a value read from the API, we should also specify `Computed: true` -- otherwise if `agent_pool_id` is changed in the API, the provider will re-create the resource and set the `agent_pool_id` to whatever was originally in your statefile. 

Normally you'll use these schema behaviors like so:
```go
// Indicates an attribute *must* be strictly managed by Terraform. If something in the API changes, Terraform
// will remove it and recreate it to match your config. 
"some_attribute": {
  Type: schema.SomeType,
  Required: true,
  ForceNew: true
}
```
or

```go
// Indicates an attribute that can either be managed by Terraform or a value read from the API. 
"some_attribute": {
  Type: schema.SomeType,
  Optional: true,
  Computed: true
}
```
I would avoid mixing `Optional/Computed` and `ForceNew` for any unintended side effects. But someone feel free to correct me if I'm wrong here 😅

Reply:
Thanks @sebasslash for the detailed explanation. I'll make the code changes.

Reply:
@sebasslash should I keep it on draft state since its in beta, or merge it and update the change log later? What do you recommend. thanks!